qiskit/synthesis/two_qubit/two_qubit_decompose.py

```diff
@@ -476,129 +240,27 @@ def __repr__(self):
 
     @classmethod
     def from_bytes(
-        cls, bytes_in: bytes, *, requested_fidelity: float, **kwargs
+        cls,
+        bytes_in: bytes,
+        *,
+        requested_fidelity: float,
+        _specialization: two_qubit_decompose.Specialization | None,
+        **kwargs,
     ) -> "TwoQubitWeylDecomposition":
         """Decode bytes into :class:`.TwoQubitWeylDecomposition`."""
         # Used by __repr__
         del kwargs  # Unused (just for display)
         b64 = base64.decodebytes(bytes_in)
         with io.BytesIO(b64) as f:
             arr = np.load(f, allow_pickle=False)
-        return cls(arr, fidelity=requested_fidelity)
+        return cls(arr, fidelity=requested_fidelity, _specialization=_specialization)
 
     def __str__(self):
         pre = f"{self.__class__.__name__}(\n\t"
         circ_indent = "\n\t".join(self.circuit(simplify=True).draw("text").lines(-1))
         return f"{pre}{circ_indent}\n)"
 
 
-class TwoQubitWeylIdEquiv(TwoQubitWeylDecomposition):
-    r""":math:`U \sim U_d(0,0,0) \sim Id`
-
-    This gate binds 0 parameters, we make it canonical by setting
-    :math:`K2_l = Id` , :math:`K2_r = Id`.
-    """
-
-    def specialize(self):
-        self.a = self.b = self.c = 0.0
-        self.K1l = self.K1l @ self.K2l
-        self.K1r = self.K1r @ self.K2r
-        self.K2l = _id.copy()
-        self.K2r = _id.copy()
-
-
-class TwoQubitWeylSWAPEquiv(TwoQubitWeylDecomposition):
-    r""":math:`U \sim U_d(\pi/4, \pi/4, \pi/4) \sim U(\pi/4, \pi/4, -\pi/4) \sim \text{SWAP}`
-
-    This gate binds 0 parameters, we make it canonical by setting
-    :math:`K2_l = Id` , :math:`K2_r = Id`.
-    """
-
-    def specialize(self):
-        if self.c > 0:
-            self.K1l = self.K1l @ self.K2r
-            self.K1r = self.K1r @ self.K2l
-        else:
-            self._is_flipped_from_original = True
-            self.K1l = self.K1l @ _ipz @ self.K2r
-            self.K1r = self.K1r @ _ipz @ self.K2l
-            self.global_phase = self.global_phase + np.pi / 2
-        self.a = self.b = self.c = np.pi / 4
-        self.K2l = _id.copy()
-        self.K2r = _id.copy()
-
-    def _weyl_gate(self, simplify, circ: QuantumCircuit, atol):
-        del self, simplify, atol  # unused
-        circ.swap(0, 1)
-        circ.global_phase -= 3 * np.pi / 4
-
-
-def _closest_partial_swap(a, b, c) -> float:
-    r"""A good approximation to the best value x to get the minimum
-    trace distance for :math:`U_d(x, x, x)` from :math:`U_d(a, b, c)`.
-    """
-    m = (a + b + c) / 3
-    am, bm, cm = a - m, b - m, c - m
-    ab, bc, ca = a - b, b - c, c - a
-
-    return m + am * bm * cm * (6 + ab * ab + bc * bc + ca * ca) / 18
-
-
-class TwoQubitWeylPartialSWAPEquiv(TwoQubitWeylDecomposition):
-    r""":math:`U \sim U_d(\alpha\pi/4, \alpha\pi/4, \alpha\pi/4) \sim \text{SWAP}^\alpha`
-    This gate binds 3 parameters, we make it canonical by setting:
-    :math:`K2_l = Id`.
```

Comment:
Maybe its worth carrying this (and the other subclass docstrings), documenting the conventions we've chosen for selecting the canonical representative for each gate into  comments somewhere in the rust implementation, since they're both partly arbitrary and fairly non-obvious?

Reply:
I added the docstring as an inline comment in: https://github.com/Qiskit/qiskit/pull/11946/commits/76d2ccf42849e29d96d46e15dd1eb58c1802a7bf